piControlIf.c

```diff
@@ -0,0 +1,582 @@
+/*!
+ * SPDX-License-Identifier: MIT
+ *
+ * Project: Pi Control
+ * Demo source code for usage of piControl driver
+ *
+ * Copyright (C) 2017 : KUNBUS GmbH, Heerweg 15C, 73370 Denkendorf, Germany
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * \file piControlIf.c
+ *
+ * \brief PI Control Interface
+ *
+ *
+ */
+
+/******************************************************************************/
+/********************************  Includes  **********************************/
+/******************************************************************************/
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include <stdio.h>
+#include <string.h>
+
+#include "piControlIf.h"
+
+#include "piControl.h"
+#ifdef KUNBUS_TEST
+#include <IoProtocol.h>
+#endif
+
+/******************************************************************************/
+/******************************  Global Vars  *********************************/
+/******************************************************************************/
+
+int PiControlHandle_g = -1;
+
+/******************************************************************************/
+/*******************************  Functions  **********************************/
+/******************************************************************************/
+
+/***********************************************************************************/
+/*!
+ * @brief Open Pi Control Interface
+ *
+ * Initialize the Pi Control Interface
+ *
+ ************************************************************************************/
+void piControlOpen(void)
+{
+        /* open handle if needed */
```

Comment:
Indention has been halved in regard to the original. Why?

Reply:
@naruxde and me decided to try and keep a unified style in the source - this is also the reason the `.editorconfig` file was introduced.
If this is not an acceptable change I can revert it.

Reply:
I leave it to the Kernel developers to decide. They (who maintained the code in the past) followed the Linux coding style rules, and I think the 8 char identation is the first rule.
https://www.kernel.org/doc/html/v6.1/process/coding-style.html